pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -963,6 +966,272 @@ func (a *ApplicationBackupController) prepareResources(
 	return nil
 }
 
+func (a *ApplicationBackupController) updateRancherProjectDetails(
+	backup *stork_api.ApplicationBackup,
+	objects []runtime.Unstructured,
+) error {
+	platformCredential, err := storkops.Instance().GetPlatformCredential(backup.Spec.PlatformCredential, backup.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	if platformCredential.Spec.Type == stork_api.PlatformCredentialRancher {
+		if err = updateRancherProjects(platformCredential, backup, objects); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func updateRancherProjects(
+	platformCredential *stork_api.PlatformCredential,
+	backup *stork_api.ApplicationBackup,
+	objects []runtime.Unstructured,
+) error {
+	projects := map[string]string{}
+	for _, o := range objects {
+		metadata, err := meta.Accessor(o)
+		if err != nil {
+			return err
+		}
+		resource := o.GetObjectKind().GroupVersionKind()
+		switch resource.Kind {
+		case "Deployment", "StatefulSet", "DeploymentConfig", "IBPPeer", "IBPCA", "IBPConsole", "IBPOrderer", "ReplicaSet":
+			err := getProjectsFromPodNamespaceSelector(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		case "NetworkPolicy":
+			err := getProjectsFromNetworkPolicy(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		case "PersistentVolume":
+			err := getProjectsFromPV(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		case "PersistentVolumeClaim":
+			err := getProjectsFromPVC(o, projects)
+			if err != nil {
+				return fmt.Errorf("error preparing %v resource %v: %v", o.GetObjectKind().GroupVersionKind().Kind, metadata.GetName(), err)
+			}
+		}
+	}
+
+	for _, namespace := range backup.Spec.Namespaces {
+		ns, err := core.Instance().GetNamespace(namespace)
+		if err != nil {
+			return err
+		}
+		for key, val := range ns.Annotations {
+			if strings.Contains(key, utils.CattleProjectPrefix) {
```

Comment:
As annotation will have both cluster:project,  where do we stip of clusterID from the annotation and extract only project?

Reply:
Currently the projectmap does not strip the clusterID.
projects[clustID:ProjectID] = projectDisplayableName
```
rancher_projects:
c-xg2ls:p-62xs2: "siva-project"

Reply:
We strip of cluster id only during the restore, `getRancherProjectMapping`, will do the stripping , Lets say `c-gw7q8:p-lmz7v` : `c-greq1:p-zgs91` is the project mapping we got from restore request. `getRancherProjectMapping` will strip the clusterID and maintain two entries one as `c-gw7q8:p-lmz7v` : `c-greq1:p-zgs91` and the other with `p-lmz7v` : `p-zgs91`, so first will be used annotation replace and the later for namespace selector replace.